src/main/kotlin/org/rust/lang/utils/Utils.kt

```diff
@@ -31,3 +32,43 @@ fun <T> using(d: Disposable, block: () -> T): T {
         d.dispose()
     }
 }
+
+/**
+ * Helper disposing [d] upon completing the execution of the [block] (under the [d])
+ *
+ * @d       Target `Disposable` to be disposed upon completion of the @block
+ * @block   Target block to be run prior to disposal of @d
+ */
+fun <D: Disposable, T> usingWith(d: D, block: (D) -> T): T {
+    try {
+        return block(d)
+    } finally {
+        d.dispose()
+    }
+}
+
+/**
+ * Helper executing the block supplied under the lock being ACQUIRED,
+ * and RELEASED upon completion (successful or not)
+ *
+ * @l       Lock to be acquired
+ * @block   Target block to be run 'under' the lock being acquired
+ */
+fun <T> lock(l: Lock, block: () -> T): T =
```

Comment:
> It would make "Lock to guard reads/updates to [cached]" comments unnecessary and would make it harder to forget to lock the lock.

It wouldn't make it any harder.